step-plans/step-plans-core/src/main/java/step/core/artefacts/reports/aggregated/AggregatedReportView.java

```diff
@@ -36,8 +40,24 @@ public String toString() {
 
     private static void recursiveToString(AggregatedReportView node, int level, StringBuilder stringBuilder) {
         String name = node.artefact.getClass().getSimpleName();
-        String indentation = new String(new char[level]).replace("\0", " ");
-        stringBuilder.append(indentation).append(name).append(": ").append(node.countTotal()).append("x\n");
-        node.children.forEach(c -> recursiveToString(c, level + 1, stringBuilder));
+        String indentation = " ".repeat(level);
+        stringBuilder.append(indentation);
+        stringBuilder.append(name).append(": ").append(node.countTotal()).append("x\n");
+        //Process children considering parent source
+        ParentSource previousParentSource = null;
+        for (AggregatedReportView childReportView: node.children) {
```

Comment:
It could be also useful to write this as a structured object as yaml or json

Reply:
not sure to get this comment

Reply:
We're now writing the report in a plain text format with a custom syntax. As a follow up task we could report this as JSON or YAML to have a clearer syntax which could be more easy to parse and handle by clients

Reply:
I get it, this is not related to this PR but a new feature request for the Aggregated report view. I'll create a separate ticket for that.